cassandra/src/main/scala/org/apache/spark/sql/cassandra/CassandraXDSourceRelation.scala

```diff
@@ -133,14 +135,34 @@ class CassandraXDSourceRelation(
   def buildScan(): RDD[Row] = baseRdd.asInstanceOf[RDD[Row]]
 
   override def buildScan(requiredColumns: Array[String], filters: Array[Filter]): RDD[Row] = {
-    val prunedRdd = maybeSelect(baseRdd, requiredColumns)
+    buildScan(requiredColumns, filters, Map.empty)
+  }
+
+  private def resolveUDFsReferences(strId: String, udfs: Map[String, NativeUDF]): Option[FunctionCallRef] =
+    udfs get(strId) map { udf =>
+      val actualParams = udf.children.collect {
+        case at: AttributeReference if(udfs contains at.toString) => Left(resolveUDFsReferences(at.toString, udfs).get)
+        case at: AttributeReference => Left(ColumnName(at.name))
+        case lit@Literal(_, DataTypes.StringType) => Right(s"$lit")
```

Comment:
case lit@Literal(_, DataTypes.StringType) => Right(s"$lit") can be covered by case lit: Literal => Right(lit.toString)